packages/financial-templates-lib/src/logger/SlackTransport.ts

```diff
@@ -186,19 +192,32 @@ class SlackHook extends Transport {
       // If the log contains a notification path then use a custom slack webhook service. This lets the transport route to
       // different slack channels depending on the context of the log.
       const webhookUrl = this.escalationPathWebhookUrls[info.notificationPath] ?? this.defaultWebHookUrl;
-
-      const payload: { blocks?: Block[]; text?: string; mrkdwn?: boolean } = { mrkdwn: this.mrkdwn };
       const layout = this.formatter(info);
-      payload.blocks = layout.blocks || undefined;
-      // If the overall payload is less than 3000 chars then we can send it all in one go to the slack API.
-      if (JSON.stringify(payload).length < SLACK_MAX_CHAR_LIMIT) {
-        await this.axiosInstance.post(webhookUrl, payload);
-      } else {
-        // Iterate over each message to send and generate a axios call for each message.
-        for (const processedBlock of processMessageBlocks(payload.blocks)) {
-          payload.blocks = processedBlock;
-          await this.axiosInstance.post(webhookUrl, payload);
-        }
+
+      // Deliver in chunks if the the combined payload is more than SLACK_MAX_CHAR_LIMIT chars.
+      const payloads: (SlackFormatterResponse & { text?: string; mrkdwn?: boolean })[] =
+        JSON.stringify(layout.blocks).length < SLACK_MAX_CHAR_LIMIT
+          ? [{ blocks: layout.blocks, mrkdwn: this.mrkdwn }]
+          : processMessageBlocks(layout.blocks).map((blocks) => {
+              return { blocks, mrkdwn: this.mrkdwn };
+            });
+
+      for (const payload of payloads) {
+        // HTTP 400: probably a badly formatted message (recoverable)
+        // HTTP > 400: indicative of a hard error (non-recoverable)
+        // https://api.slack.com/messaging/webhooks#handling_errors
+        await this.axiosInstance.post(webhookUrl, payload).catch(async (err) => {
+          if (err?.["isAxiosError"]) {
+            const code = err?.["response"]?.["status"];
+            if (code !== 400) {
+              throw err; // Probably a hard error -> escalate.
+            }
+            const dropped = JSON.stringify(payload).slice(0, SLACK_MAX_CHAR_LIMIT - 1000);
+            const warning = { at: "SlackHook::log", level: "warn", message: "Dropped log message!", dropped };
```

Comment:
Discussed with Reinis: it might be a problem that `notificationPath` is not included here, since that's used much more heavily on the UMA side.

Reply:
Another thing to consider: emitting a log message here is strictly a layering violation. The motivation for doing it here is that this is where we have the necessary context to try a resubmission, with knowledge that the payload we submitted was probably invalid and must therefore be transformed in some way.